dali/operators/audio/nonsilence_op.h

```diff
@@ -43,6 +43,17 @@ struct Args {
 };
 
 
+/**
+ * If the buffer is not silent, add window length to the actual result,
+ * since we don't know where in the window the non-silent signal is
+ */
+void extend_nonsilent_range(std::pair<int, int> &thresholding_result, int window_length) {
```

Comment:
this is not 100% correct. We should be picking the end of the window containing the end of the range and for that you need to involve both the `window_length` and the `window_step`:

```
auto full_steps = res.second / window_step
if (res.second % window_step) {  // the end of the range is at the beginning of a window
  full_steps--;
}
return full_steps * window_step + window_length;  // end of the window containing `res.second`
```